### DISCUSSION TITLE
Shairport service dies upon connection, but it runs well when launched as a command

### DISCUSSION THREAD

**Discussion by csarmiento** [mikebrady/shairport-sync#1681] [Q&A] [answered]
Hey! 

First of all some context:

Running on: Raspberry Pi 3 (`Linux 5.10.103-v7+ #1529 SMP Tue Mar 8 12:21:37 GMT 2022 armv7l GNU/Linux`)
Installed following: https://github.com/mikebrady/shairport-sync/blob/master/BUILD.md

**Problem**

I am running in a strange situation where the `shairport-sync` service dies upon any connection from any of my Apple devices... The logs I see are:

```
Jun  2 11:53:50 music shairport-sync[480]: Failed to create secure directory (/home/shairport-sync/.config/pulse): No such file or directory
Jun  2 11:53:50 music shairport-sync[480]: ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
Jun  2 11:53:50 music shairport-sync[480]: ALSA lib conf.c:4568:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
Jun  2 11:53:50 music shairport-sync[480]: ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
Jun  2 11:53:50 music shairport-sync[480]: ALSA lib conf.c:4568:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
Jun  2 11:53:50 music shairport-sync[480]: ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
Jun  2 11:53:50 music shairport-sync[480]: ALSA lib conf.c:4568:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
Jun  2 11:53:50 music shairport-sync[480]: ALSA lib conf.c:5047:(snd_config_expand) Evaluate error: No such file or directory
Jun  2 11:53:50 music shairport-sync[480]: ALSA lib pcm.c:2565:(snd_pcm_open_noupdate) Unknown PCM default
Jun  2 11:53:50 music shairport-sync: fatal error: An unrecoverable error, "output_device_error_2", has been detected. Doing an emergency exit, as no run_this_if_an_unfixable_error_is_detected program.
Jun  2 11:53:50 music systemd[1]: shairport-sync.service: Succeeded.
```

**But**, when I execute `shairport-sync --log-to-syslog` from the command line logged as `pi` user, it runs well, no problems at all...

Maybe someone knows if there is some sort of configuration issue going on?

Thanks.

**Comment by mikebrady** [comment, marked as answer]
Thanks for the post, though it's probably more of an issue than a discussion item. At a guess, your system has a GUI and it using PulseAudio or PipeWire as a sound server. If so, then perhaps [this](https://github.com/mikebrady/shairport-sync/blob/master/ADVANCED%20TOPICS/PulseAudioAndPipeWire.md) might help.

**Reply by csarmiento** [reply]
Thank you, it is correct. I had to reinstall the OS using a Lite version as I want it to be running as a service. However, I had to disable the built in sound card, as it seemed to be the one used by default, and I wasn't able to make it work through the `/etc/shairport-sync.conf`. The procedure I followed is the one in this answer: https://superuser.com/a/1045885/227240

Thanks a lot for your effort with **shairport-sync**!
